server/controllers/UserController.js

```diff
@@ -32,22 +45,93 @@ class UserController {
           new: true, // Return the updated user
         },
       )
-      return updatedUser
+      return updatedUser.toObject()
     } catch (error) {
       console.error('Error while updating user:', error)
       throw error
     }
   }
 
-  async deleteUser(userId) {
+  async delete(userId) {
     try {
       const deletedUser = await UserModel.findByIdAndRemove(userId)
-      return deletedUser
+      return deletedUser.toObject()
     } catch (error) {
       console.error('Error while deleting user:', error)
       throw error
     }
   }
+
+  async register(userData) {
+    const userWithEmail = await this.getByEmail(userData.email)
+    if (userWithEmail != null) {
+      const error = new Error('Email already exists')
+      error.statusCode = 400
+      throw error
+    }
+    const salt = bcrypt.genSaltSync(10)
+    const hashedPassword = bcrypt.hashSync(userData.password, salt)
+    const newlyCreatedUser = await this.create({
+      username: userData.username,
+      email: userData.email,
+      password: hashedPassword,
+    })
+
+    const accessToken = this.createAccessToken(newlyCreatedUser)
+    const userDTO = {
+      id: newlyCreatedUser._id,
+      username: newlyCreatedUser.username,
+      email: newlyCreatedUser.email,
+      accessToken,
+    }
+    return userDTO
+  }
+
+  createAccessToken(userData) {
+    try {
+      const accessToken = jwt.sign(
+        {
+          id: userData._id,
+          username: userData.username,
+          email: userData.email,
+        },
+        config.server.jwtSecret,
+      )
+      return accessToken
+    } catch (error) {
+      console.error('Error with creating token')
+      throw new Error('Could not create token: ', error)
```

Comment:
On my machine, when passing in 2 arguments to error, the 2nd argument doesnt get sent over to the frontend. Consider this?

```suggestion
      throw new Error(`Could not create token: ${error}`)
```